MNE-Python failing test_io_set from mne.io.eeglab.tests.test_eeglab

Hello, I am also getting an error in the same test as issue  #4855 (but a different error, and in  version 0.16) : 

`    
    ================ FAILURES  ================
    _______________    test_io_set   _______________

    args = (), kwargs = {}
    def skipper_func(*args, **kwargs):
        """Skipper for normal test functions."""
        if skip_val():
            raise SkipTest(get_msg(f, msg))
        else:
>           return f(*args, **kwargs)

/user/ngayraud/home/anaconda2/lib/python2.7/site-packages/numpy/testing/decorators.py:147: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
mne/io/eeglab/tests/test_eeglab.py:234: in test_io_set
    np.array([0., 0., 0.]))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

x_id = array([ True,  True,  True], dtype=bool), y_id = array([False, False, False], dtype=bool), hasval = 'nan'

    def chk_same_position(x_id, y_id, hasval='nan'):
        """Handling nan/inf: check that x and y have the nan/inf at the same
            locations."""
        try:
            assert_array_equal(x_id, y_id)
        except AssertionError:
            msg = build_err_msg([x, y],
                                err_msg + '\nx and y %s location mismatch:'
                                % (hasval), verbose=verbose, header=header,
                                names=('x', 'y'), precision=precision)
>           raise AssertionError(msg)
E           AssertionError: 
E           Arrays are not equal
E           
E           x and y nan location mismatch:
E            x: array([ nan,  nan,  nan])
E            y: array([ 0.,  0.,  0.])

/user/ngayraud/home/anaconda2/lib/python2.7/site-packages/numpy/testing/utils.py:704: AssertionError`

I checked the solution proposed in MRG: Fix loc #4860, where all `[0., 0., 0.]` are replaced by `np.nan`.

It seems that the test passes when the line 231-232

`
assert_array_equal(raw.info['chs'][i]['loc'][:3],
                            np.array([0., 0., 0.]))
`

is replaced by

`assert_array_equal(raw.info['chs'][i]['loc'][:3], [np.nan] * 3)`

--- a/.travis.yml
+++ b/.travis.yml
@@ -26,11 +26,12 @@ matrix:
           env: CONDA_ENVIRONMENT="environment.yml"
                SPLIT=1
 
+        # OSX
         - os: osx
-         env: CONDA_ENVIRONMENT="environment.yml"
+          env: CONDA_ENVIRONMENT="environment.yml"
               SPLIT=0
         - os: osx
-         env: CONDA_ENVIRONMENT="environment.yml"
+          env: CONDA_ENVIRONMENT="environment.yml"
               SPLIT=1
 
         # 2.7 + non-default stim channel

--- a/.travis.yml
+++ b/.travis.yml
@@ -160,7 +160,11 @@ script:
         fi;
         USE_DIRS="";
         for DIR in ${MNE_DIRS}; do
-          USE_DIRS="mne/${DIR}/tests ${USE_DIRS}";
+          if [ "${DIR}" == "." ]; then
+            USE_DIRS="mne/tests ${USE_DIRS}";
+          else
+            USE_DIRS="mne/${DIR}/ ${USE_DIRS}";
+          fi;
         done;
       fi;
     - echo py.test -m "${CONDITION}" ${OPTION} ${USE_DIRS} --cov=mne;

--- a/.travis.yml
+++ b/.travis.yml
@@ -161,7 +161,7 @@ script:
         USE_DIRS="";
         for DIR in ${MNE_DIRS}; do
           if [ "${DIR}" == "." ]; then
-            USE_DIRS="mne/tests ${USE_DIRS}";
+            USE_DIRS="mne/tests mne/*.py ${USE_DIRS}";
           else
             USE_DIRS="mne/${DIR}/ ${USE_DIRS}";
           fi;

--- a/.travis.yml
+++ b/.travis.yml
@@ -26,13 +26,12 @@ matrix:
           env: CONDA_ENVIRONMENT="environment.yml"
                SPLIT=1
 
-        # OSX (as of 2018/01/19, fails to run/install 90% of the time, and takes hours to do so)
-        #- os: osx
-        #  env: CONDA_ENVIRONMENT="environment.yml"
-        #       SPLIT=0
-        #- os: osx
-        #  env: CONDA_ENVIRONMENT="environment.yml"
-        #       SPLIT=1
+        - os: osx
+         env: CONDA_ENVIRONMENT="environment.yml"
+              SPLIT=0
+        - os: osx
+         env: CONDA_ENVIRONMENT="environment.yml"
+              SPLIT=1
 
         # 2.7 + non-default stim channel
         - os: linux

--- a/mne/inverse_sparse/mxne_optim.py
+++ b/mne/inverse_sparse/mxne_optim.py
@@ -72,10 +72,10 @@ def prox_l21(Y, alpha, n_orient, shape=None, is_stft=False):
     >>> Y = np.tile(np.array([0, 4, 3, 0, 0], dtype=np.float), (2, 1))
     >>> Y = np.r_[Y, np.zeros_like(Y)]
     >>> print(Y)
-    [[0. 4. 3. 0. 0.]
-     [0. 4. 3. 0. 0.]
-     [0. 0. 0. 0. 0.]
-     [0. 0. 0. 0. 0.]]
+    [[ 0.  4.  3.  0.  0.]
+     [ 0.  4.  3.  0.  0.]
+     [ 0.  0.  0.  0.  0.]
+     [ 0.  0.  0.  0.  0.]]
     >>> Yp, active_set = prox_l21(Y, 2, 2)
     >>> print(Yp)  # doctest:+SKIP
     [[0.         2.86862915 2.15147186 0.         0.        ]
@@ -146,10 +146,10 @@ def prox_l1(Y, alpha, n_orient):
     >>> Y = np.tile(np.array([1, 2, 3, 2, 0], dtype=np.float), (2, 1))
     >>> Y = np.r_[Y, np.zeros_like(Y)]
     >>> print(Y)
-    [[1. 2. 3. 2. 0.]
-     [1. 2. 3. 2. 0.]
-     [0. 0. 0. 0. 0.]
-     [0. 0. 0. 0. 0.]]
+    [[ 1.  2.  3.  2.  0.]
+     [ 1.  2.  3.  2.  0.]
+     [ 0.  0.  0.  0.  0.]
+     [ 0.  0.  0.  0.  0.]]
     >>> Yp, active_set = prox_l1(Y, 2, 2)
     >>> print(Yp)  # doctest:+SKIP
     [[0.         0.58578644 1.58578644 0.58578644 0.        ]

--- a/mne/inverse_sparse/mxne_optim.py
+++ b/mne/inverse_sparse/mxne_optim.py
@@ -71,7 +71,7 @@ def prox_l21(Y, alpha, n_orient, shape=None, is_stft=False):
     -------
     >>> Y = np.tile(np.array([0, 4, 3, 0, 0], dtype=np.float), (2, 1))
     >>> Y = np.r_[Y, np.zeros_like(Y)]
-    >>> print(Y)
+    >>> print(Y)  # doctest:+SKIP
     [[ 0.  4.  3.  0.  0.]
      [ 0.  4.  3.  0.  0.]
      [ 0.  0.  0.  0.  0.]
@@ -145,7 +145,7 @@ def prox_l1(Y, alpha, n_orient):
     -------
     >>> Y = np.tile(np.array([1, 2, 3, 2, 0], dtype=np.float), (2, 1))
     >>> Y = np.r_[Y, np.zeros_like(Y)]
-    >>> print(Y)
+    >>> print(Y)  # doctest:+SKIP
     [[ 1.  2.  3.  2.  0.]
      [ 1.  2.  3.  2.  0.]
      [ 0.  0.  0.  0.  0.]

--- a/mne/io/eeglab/tests/test_eeglab.py
+++ b/mne/io/eeglab/tests/test_eeglab.py
@@ -230,7 +230,6 @@ def test_io_set():
         assert_equal(raw.info['chs'][i]['ch_name'], ch_names[i])
         assert_array_equal(raw.info['chs'][i]['loc'][:3],
                            np.array([np.nan, np.nan, np.nan]))
-    1 / 0  # BOOM!
 
 
 @testing.requires_testing_data

--- a/mne/io/eeglab/tests/test_eeglab.py
+++ b/mne/io/eeglab/tests/test_eeglab.py
@@ -229,7 +229,7 @@ def test_io_set():
     for i in range(3):
         assert_equal(raw.info['chs'][i]['ch_name'], ch_names[i])
         assert_array_equal(raw.info['chs'][i]['loc'][:3],
-                           np.array([0., 0., 0.]))
+                           np.array([np.nan, np.nan, np.nan]))
 
 
 @testing.requires_testing_data

--- a/mne/io/eeglab/tests/test_eeglab.py
+++ b/mne/io/eeglab/tests/test_eeglab.py
@@ -230,6 +230,7 @@ def test_io_set():
         assert_equal(raw.info['chs'][i]['ch_name'], ch_names[i])
         assert_array_equal(raw.info['chs'][i]['loc'][:3],
                            np.array([np.nan, np.nan, np.nan]))
+    1/0  # BOOM!
 
 
 @testing.requires_testing_data

--- a/mne/io/eeglab/tests/test_eeglab.py
+++ b/mne/io/eeglab/tests/test_eeglab.py
@@ -230,7 +230,7 @@ def test_io_set():
         assert_equal(raw.info['chs'][i]['ch_name'], ch_names[i])
         assert_array_equal(raw.info['chs'][i]['loc'][:3],
                            np.array([np.nan, np.nan, np.nan]))
-    1/0  # BOOM!
+    1 / 0  # BOOM!
 
 
 @testing.requires_testing_data

